boards/components/src/debug_writer.rs

```diff
@@ -19,24 +19,53 @@
 // Last modified: 11/07/2019
 
 use capsules::virtual_uart::{MuxUart, UartDevice};
+use core::mem::MaybeUninit;
 use kernel::capabilities;
 use kernel::collections::ring_buffer::RingBuffer;
 use kernel::component::Component;
 use kernel::hil;
 use kernel::hil::uart;
-use kernel::static_init;
 
 // The sum of the output_buf and internal_buf is set to a multiple of 1024 bytes in order to avoid excessive
 // padding between kernel memory and application memory (which often needs to be aligned to at
 // least a 1 KiB boundary). This is not _semantically_ critical, but helps keep buffers on 1 KiB
 // boundaries in some cases. Of course, these definitions are only advisory, and individual boards
 // can choose to pass in their own buffers with different lengths.
-const DEBUG_BUFFER_KBYTE: usize = 1;
+pub const DEBUG_BUFFER_KBYTE: usize = 1;
 
 // Bytes [0, DEBUG_BUFFER_SPLIT) are used for output_buf while bytes
 // [DEBUG_BUFFER_SPLIT, DEBUG_BUFFER_KBYTE * 1024) are used for internal_buf.
 const DEBUG_BUFFER_SPLIT: usize = 64;
 
+#[macro_export]
+macro_rules! debug_writer_component_static {
+    () => {{
+        use components::debug_writer::DEBUG_BUFFER_KBYTE;
```

Comment:
```suggestion
        use $crate::debug_writer::DEBUG_BUFFER_KBYTE;
```
This is the idiomatic way to do this, since it does not depend on whatever board is using the macro having `components` in scope and using the same name for the import.